css-view-transitions-2/Overview.bs

```diff
@@ -593,50 +620,57 @@ When capturing the old or new state for an element, perform the following steps
 			Note: this means that any running transition would be skipped when the document is ready
 			to unload.
 
-		1. Set |oldDocument|'s [=auto-skip view transitions=] to true.
-
-			Note: this means that calling {{Document/startViewTransition()}} while capturing the old document for a cross-document view-transition would run the callback but skip the animation.
-
 		1. Let |outboundTransition| be a new {{ViewTransition}} object in |oldDocument|'s [=relevant Realm=],
-			whose [=ViewTransition/active types=] is |resolvedRule|, and whose [=ViewTransition/process old state captured=] is set to the following steps:
+			whose [=ViewTransition/active types=] is |resolvedRule|.
 
-			Issue: should we check for the opt-in again, in case there was a CSSOM change in a requestAnimationFrame callback?
+		1. Set |outboundTransition|'s [=is outbound cross-document transition=] to true.
 
-			1. If |outboundTransition|'s [=ViewTransition/phase=] is "`done`", then call |onReady| and return.
+		1. Run the following steps when [=Perform pending transition operations|performing pending transition operations=]:
 
 			1. Assert: |outboundTransition|'s  [=ViewTransition/phase=] is "`pending-capture`".
 
-			1. [=Clear view transition=] |outboundTransition|.
+			1. [=Capture the old state=] for |outboundTransition|.
 
-				Note: The ViewTransition object on the old Document should be destroyed after its state has been copied to the new Document below.
-					We explicitly clear it here since the old Document may be cached by the UA.
+			1. If this succeeded, then:
+
+				1. Let |clonedNamedElements| be |outboundTransition|'s [=ViewTransition/named elements=] to a [=map/clone=] of |outboundTransition|'s [=ViewTransition/named elements=].
+
+				1. Let |initialSnapshotContainingBlockSize| to |outboundTransition|'s [=ViewTransition/initial snapshot containing block size=].
+
+				1. Set |newDocument|'s [=inbound view transition steps=] to the following steps:
```

Comment:
I actually find it cleaner to stash initialSnapshotContainingBlockSize and clonedNamedElements on the Document and inline these steps into "resolve cross-document view-transition", where they run. Instead of binding them to a callback and then stashing that on the new Document.